packages/common/src/store/pages/chat/slice.ts

```diff
@@ -372,50 +373,52 @@ const slice = createSlice({
         return
       }
 
+      // Return early if we've seen this message
       const existingMessage = getMessage(
         state.messages[chatId],
         message.message_id
       )
-      if (!existingMessage) {
-        chatMessagesAdapter.addOne(state.messages[chatId], {
-          ...message,
-          hasTail: true,
-          status: status ?? Status.IDLE
-        })
-        chatsAdapter.updateOne(state.chats, {
-          id: chatId,
-          changes: {
-            last_message: message.message,
-            last_message_at: message.created_at,
-            // If a new message comes through, we don't need to recheck permissions anymore
-            recheck_permissions: false
-          }
-        })
-        recalculatePreviousMessageHasTail(state.messages[chatId], 0)
-      }
-    },
-    incrementUnreadCount: (
-      state,
-      action: PayloadAction<{ chatId: string }>
-    ) => {
-      const { chatId } = action.payload
-      // If we're actively reading, this will immediately get marked as read.
-      // Ignore the unread bump to prevent flicker
-      if (state.activeChatId !== chatId) {
-        const existingChat = getChat(state, chatId)
-        const optimisticRead = state.optimisticChatRead[chatId]
-        const existingUnreadCount = optimisticRead
-          ? optimisticRead.unread_message_count
-          : existingChat?.unread_message_count ?? 0
-        chatsAdapter.updateOne(state.chats, {
-          id: chatId,
-          changes: { unread_message_count: existingUnreadCount + 1 }
-        })
-      }
-      if (state.optimisticUnreadMessagesCount) {
-        state.optimisticUnreadMessagesCount += 1
-      } else {
-        state.optimisticUnreadMessagesCount = state.unreadMessagesCount + 1
+      if (existingMessage) return
+
+      // Add the message
+      chatMessagesAdapter.addOne(state.messages[chatId], {
+        ...message,
+        hasTail: true,
+        status: status ?? Status.IDLE
+      })
+      chatsAdapter.updateOne(state.chats, {
+        id: chatId,
+        changes: {
+          last_message: message.message,
+          last_message_at: message.created_at,
+          // If a new message comes through, we don't need to recheck permissions anymore
+          recheck_permissions: false
+        }
+      })
+
+      // Recalculate tails
+      recalculatePreviousMessageHasTail(state.messages[chatId], 0)
+
+      // Handle unread counts
+
+      if (!isSelfMessage) {
+        // If we're actively reading, this will immediately get marked as read.
+        // Ignore the unread bump to prevent flicker
+        if (state.activeChatId !== chatId) {
+          const existingChat = getChat(state, chatId)
+          const optimisticRead = state.optimisticChatRead[chatId]
```

Comment:
Were we also gonna delete the optimistic unread count in some case? I think it was when a new non-self message comes in?

Reply:
I think we don't need to do this